services.py

```
@@ -0,0 +1,368 @@
+        try:
+            scenario = Scenario.query.get(scenario_id)
+            if not scenario:
+                raise NotFoundError("シナリオ", str(scenario_id))
+            return scenario
+        except SQLAlchemyError as e:
+            logger.error(f"シナリオ取得エラー (ID: {scenario_id}): {e}")
+            raise AppError(
+                message="シナリオの取得中にデータベースエラーが発生しました",
+                code="DATABASE_ERROR",
+                status_code=500
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)